src/List/ListItemText.js

```diff
@@ -20,6 +20,11 @@ export const styleSheet = createStyleSheet('MuiListItemText', () => {
         paddingLeft: 56,
       },
     },
+    dense: {
+      '& > *': {
```

Comment:
So, what do you think about @mbrookes's idea.
1. We set the font size to the parent element
2. We add a class on the Text to get the inheritance.

The win is a flatten CSS selector, that's always better for composition.